hamp_radar/geometries.py

```diff
@@ -75,3 +77,49 @@ class MultiMainBlockGeometry:
     count: int
     step: Optional[int]
     subblocks: List[SingleSubBlockGeometry]
+
+
+@dataclass
+class DatasetBlockGeometry:
+    """
+    Representation of a block of a dataset is a multimainblock with
+    the name of the .pds file it's sourced from.
+
+    Attributes:
+        filename (Path): The name of the .pds file the mmbg is from.
+        mainblock (MultiMainBlockGeometry): The mmbg in the file.
+    """
+
+    filename: Path
+    mainblock: MultiMainBlockGeometry
+
+
+@dataclass
+class DatasetGeometry:
+    """
+    Representation of a dataset's geometry is a DatasetBlockGeometry instance
+    for the PPAR and a list of DatasetBlockGeometry instances
+    for the data associated with that DSP configuration.
+
+    Attributes:
+        ppar (DatasetBlockGeometry): The block for the PPAR in the file.
+        data (List[DatasetBlockGeometry]): The blocks for the data associated with that PPAR.
+    """
+
+    ppar: DatasetBlockGeometry
+    data: List[DatasetBlockGeometry]
+
+
+@serde
+class FlightGeometry:
```

Comment:
I'm not yet super sure if `FlightGeometry` is the best abstraction for this thing. My main issue would be, that this forces us to attach a `name` to a collection of `DatasetGeometry`s. But as I also don't really know that much better at the moment, I'd probably keep it.

Reply:
Yes I'm not sure because really it's not necessarily a flight... but then again the current creation of FlightGeometry ensures the collection contains at least 1 ppar and so maybe it is not just a collection. I don't think it should be called "Flight" Geometry but I haven't thought of something better. We can discuss if you like...